### DISCUSSION TITLE
Authenticate users with Google One-Tap

### DISCUSSION THREAD

**Discussion by SinaBYR** [pocketbase/pocketbase#2963] [Q&A]
Is there a way to manually create new user who is authenticated with Google One-Tap? Basically what Google One-Tap does is show you a pop-up window where user's signed-in google accounts show up, then they click on their desired google account to use that account to sign into the website. They also let you have a "onSuccess" handler to basically handle when user's data is successfully returned via Google API. The main problem is that Pocketbase does not expose a custom method to insert the returned account information to the "users" table to create a new google-authenticated user.

Here's what I'm trying to accomplish in a React-ish way:
```tsx
<GoogleLogin
   onSuccess={response => {
      const userData = jwtDecode(response)
      //  newlyCreatedUserData is of type "RecordAuthResponse<Record>"
      const newlyCreatedUserData = pb.collection('users').somePocketBaseCustomMethod(userData)
      console.log(newlyCreatedUserData)
   }}
   useOneTap
/>
```

**Comment by ganigeorgiev** [comment]
This is too vendor specific and there are no plans to have it integrated in PocketBase.

The current [OAuth2 "all-in-one" flow](https://pocketbase.io/docs/authentication/#oauth2-integration) is almost identical to the described one from your example (at least in web).

> The main problem is that Pocketbase does not expose a custom method to insert the returned account information to the "users" table to create a new google-authenticated user.

You can register your own [custom route](https://pocketbase.io/docs/router/#register-new-route) which could perform the necessary db operations that are required. With the upcoming v0.17.0 release you'll be able to do that also with JS (_the "Use as framework" guides will be also updated with more details and examples_).
